### ISSUE TITLE
Frontline units appear to be fielded by order of purchase

### ISSUE THREAD

**Issue by Starfire13** [dcs-liberation/dcs_liberation#2395]
### Affected versions

Development build

### Development build

https://github.com/dcs-liberation/dcs_liberation/actions/runs/2901434554

### Description

After Sandman reported on Discord that he had trouble with frontline force composition, I decided to run a test on 6.x, and it appears that frontline units are in fact fielded by order of purchase. It's possible that this is intended behaviour, but it seems odd to have it this way rather than simply picking a random assortment of vehicles.

To clarify what I am encountering, let's say I have a frontline that accepts 51 vehicles. I buy 50 Abrams on Turn 0. Then 50 Strykers on the next turn. On Turn 2 (the airfield has a factory so I now have 100 vehicles available for the frontline - 50 Abrams and 50 Strykers), there will be 50 Abrams and 1 Stryker on the frontline.

### Save game and other files

[North Russia #4236 Turn 4.zip](https://github.com/dcs-liberation/dcs_liberation/files/9438143/North.Russia.4236.Turn.4.zip)
[liberation_nextturn.zip](https://github.com/dcs-liberation/dcs_liberation/files/9438144/liberation_nextturn.zip)

**Comment by DanAlbert** [comment]
Pretty sure this is just an underinvested area and no one has ever tried to make it smarter than this.

Rather than random, it should fill according to the unit balance that it uses when buying units.

**Comment by Starfire13** [comment]
Fair enough. As long as we know how it works, we can work around it for now.
